### DISCUSSION TITLE
Predict with a selected modle

### DISCUSSION THREAD

**Discussion by xinwei-sher** [mljar/mljar-supervised#421] [Ideas]
The predict function uses the best model from AutoML. I'm interested to see the predictions from other models as well. Is there any way to select one of the other models and use it for prediction? Thanks.

"predict(self, X)
Computes predictions from AutoML best model."

**Comment by pplonski** [comment]
To compute predictions from selected model you need to change the model name in the `best_model.txt` file, then reload the AutoML and compute predictions. It is very ugly approach. I've added issue to add such functionality to select model for predictions: https://github.com/mljar/mljar-supervised/issues/423

**Reply by elcolie** [reply]
I don't have `best_model.txt`. Where I can find it?

**Reply by elcolie** [reply]
I think I found it. It is `params.json`. Then looking for `best_model`

**Reply by pplonski** [reply]
@elcolie you are right, it is in `params.json` with key `best_model`. Sorry for confusion. Please let me know if it works.

**Reply by xinwei-sher** [reply]
This doesn't work.

I changed the key best_model in params.json to another model, "2_Default_LightGBM", and tried this

automl = AutoML(results_path =  results_path)
automl.fit(X_train, y_train)
automl.predict(X_test)

This is the message I got:
This model has already been fitted. You can use predict methods or select a new 'results_path' for a new 'fit()'.
2021-07-09 19:29:01,810 supervised.exceptions ERROR This model has not been fitted yet. Please call `fit()` first.

---------------------------------------------------------------------------
AutoMLException                           Traceback (most recent call last)
<ipython-input-53-42f182ac8a72> in <module>
----> 1 automl.predict(X_test)

~/miniconda3/envs/autogluon/lib/python3.7/site-packages/supervised/automl.py in predict(self, X)
    354             AutoMLException: Model has not yet been fitted.
    355         """
--> 356         return self._predict(X)
    357 
    358     def predict_proba(self, X):

~/miniconda3/envs/autogluon/lib/python3.7/site-packages/supervised/base_automl.py in _predict(self, X)
   1304     def _predict(self, X):
   1305 
-> 1306         predictions = self._base_predict(X)
   1307         # Return predictions
   1308         # If classification task the result is in column 'label'

~/miniconda3/envs/autogluon/lib/python3.7/site-packages/supervised/base_automl.py in _base_predict(self, X, model)
   1241         if model is None:
   1242             raise AutoMLException(
-> 1243                 "This model has not been fitted yet. Please call `fit()` first."
   1244             )
   1245 

AutoMLException: This model has not been fitted yet. Please call `fit()` first.

**Reply by pplonski** [reply]
This discussion is connected with https://github.com/mljar/mljar-supervised/issues/423

I will try to fix this soon. Apologize for problems.
